internal/utils/main.go

```diff
@@ -713,3 +743,21 @@ func LedgerEntryToLedgerKeyHash(ledgerEntry xdr.LedgerEntry) string {
 
 	return ledgerKeyHash
 }
+
+// CreateLedgerBackend creates a ledger backend using captive core or datastore
+// Defaults to using datastore
+func CreateLedgerBackend(ctx context.Context, useCaptiveCore bool, env EnvironmentDetails) (ledgerbackend.LedgerBackend, error) {
+	if useCaptiveCore {
+		backend, err := env.CreateCaptiveCoreBackend()
+		if err != nil {
+			return nil, err
+		}
+		return backend, nil
+	}
+
+	backend, err := ledgerbackend.NewCloudStorageBackend(ctx, env.StorageURL)
```

Comment:
is this method going to be agnostic to cloud provider? or will each cloud provider have its own backend?

Reply:
Yes it is cloud provider agnostic. It's just only the GCS datastore is defined right now